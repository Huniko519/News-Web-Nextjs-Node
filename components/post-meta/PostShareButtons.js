import React from 'react';
import PropTypes from 'prop-types';

const PostShareButtons = (props) => {
  const { post, dataTrack } = props;
  const socialTitle = post.social_title?.rendered ? post.social_title.rendered : post.title.raw;
  const dimensions = {
    width: post?.template === 'opinion' ? 20 : 28,
    height: post?.template === 'opinion' ? 20 : 28,
  };

  return (
    <>
      <div className="inews__post-share">
        <amp-social-share
          data-track={`email_${dataTrack}`}
          type="email"
          data-param-subject={socialTitle}
          data-param-body={`${post.link}?ito=email_${dataTrack}`}
          aria-label="share via email"
          {...dimensions}
        />
        <amp-social-share
          data-track={`twitter_${dataTrack}`}
          type="twitter"
          data-param-url={`${post.link}?ito=twitter_${dataTrack}`}
          data-param-text={socialTitle}
          aria-label="share via twitter"
          {...dimensions}
        />
        <amp-social-share
          data-track={`whatsapp_${dataTrack}`}
          type="whatsapp"
          data-param-text={`${socialTitle} - ${post.link}?ito=whatsapp_${dataTrack}`}
          aria-label="share via whatsapp"
          {...dimensions}
        />
        <amp-social-share
          data-track={`facebook_${dataTrack}`}
          type="facebook"
          data-param-href={`${post.link}?ito=facebook_${dataTrack}`}
          data-param-quote={socialTitle}
          aria-label="share via facebook"
          data-param-app_id="817222585498956"
          {...dimensions}
        />
      </div>
    </>
  );
};

PostShareButtons.propTypes = {
  post: PropTypes.objectOf(PropTypes.any).isRequired,
  dataTrack: PropTypes.string.isRequired,
};

export default PostShareButtons;
