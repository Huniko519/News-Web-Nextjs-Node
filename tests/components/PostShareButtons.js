import React from 'react';
import PostShareButtons from '../../components/post-meta/PostShareButtons';

const post = {
  id: 1,
  title: {
    rendered: 'title',
    raw: 'title',
  },
  social_title: {
    rendered: "Social ' Title",
  },
  link: 'https://inews-article.com/test-article',
};
const dataTrack = 'track';

test('renders the snapshot', () => {
  const postShareButtons = shallow(<PostShareButtons post={post} dataTrack={dataTrack} />);
  expect(postShareButtons).toMatchSnapshot();
});

it('correctly configures the Email <amp-social-share /> for element', () => {
  const postShareButtons = shallow(<PostShareButtons post={post} dataTrack={dataTrack} />);

  const share = postShareButtons.find('amp-social-share[type="email"]');
  expect(share.length).toBe(1);

  expect(share.prop('data-track')).toBe(`email_${dataTrack}`);
  expect(share.prop('data-param-body')).toBe(`${post.link}?ito=email_${dataTrack}`);
  expect(share.prop('data-param-subject')).toBe("Social ' Title");
});

it('correctly configures the Facebook <amp-social-share /> for element', () => {
  const postShareButtons = shallow(<PostShareButtons post={post} dataTrack={dataTrack} />);

  const share = postShareButtons.find('amp-social-share[type="facebook"]');
  expect(share.length).toBe(1);

  expect(share.prop('data-track')).toBe(`facebook_${dataTrack}`);
  expect(share.prop('data-param-href')).toBe(`${post.link}?ito=facebook_${dataTrack}`);
  expect(share.prop('data-param-quote')).toBe("Social ' Title");
  expect(share.prop('data-param-app_id')).toBe('817222585498956');
});

it('correctly configures the Twitter <amp-social-share /> for element', () => {
  const postShareButtons = shallow(<PostShareButtons post={post} dataTrack={dataTrack} />);

  const share = postShareButtons.find('amp-social-share[type="twitter"]');
  expect(share.length).toBe(1);

  expect(share.prop('data-track')).toBe(`twitter_${dataTrack}`);
  expect(share.prop('data-param-url')).toBe(`${post.link}?ito=twitter_${dataTrack}`);
  expect(share.prop('data-param-text')).toBe("Social ' Title");
});

it('correctly configures the Whatsapp <amp-social-share /> for element', () => {
  const postShareButtons = shallow(<PostShareButtons post={post} dataTrack={dataTrack} />);

  const share = postShareButtons.find('amp-social-share[type="whatsapp"]');
  expect(share.length).toBe(1);

  expect(share.prop('data-track')).toBe(`whatsapp_${dataTrack}`);
  expect(share.prop('data-param-text')).toBe(`Social ' Title - ${post.link}?ito=whatsapp_${dataTrack}`);
});
