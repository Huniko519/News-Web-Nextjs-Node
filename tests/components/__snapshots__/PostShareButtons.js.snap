// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<Fragment>
  <div
    className="inews__post-share"
  >
    <amp-social-share
      aria-label="share via email"
      data-param-body="https://inews-article.com/test-article?ito=email_track"
      data-param-subject="Social ' Title"
      data-track="email_track"
      height={28}
      type="email"
      width={28}
    />
    <amp-social-share
      aria-label="share via twitter"
      data-param-text="Social ' Title"
      data-param-url="https://inews-article.com/test-article?ito=twitter_track"
      data-track="twitter_track"
      height={28}
      type="twitter"
      width={28}
    />
    <amp-social-share
      aria-label="share via whatsapp"
      data-param-text="Social ' Title - https://inews-article.com/test-article?ito=whatsapp_track"
      data-track="whatsapp_track"
      height={28}
      type="whatsapp"
      width={28}
    />
    <amp-social-share
      aria-label="share via facebook"
      data-param-app_id="817222585498956"
      data-param-href="https://inews-article.com/test-article?ito=facebook_track"
      data-param-quote="Social ' Title"
      data-track="facebook_track"
      height={28}
      type="facebook"
      width={28}
    />
  </div>
</Fragment>
`;
